pkg/loki/loki.go

```diff
@@ -140,37 +141,33 @@ func New(cfg Config) (*Loki, error) {
 }
 
 func (t *Loki) setupAuthMiddleware() {
+	t.cfg.Server.GRPCMiddleware = []grpc.UnaryServerInterceptor{serverutil.RecoveryGRPCUnaryInterceptor}
```

Comment:
I'm not sure if we want recovery middleware everywhere -- what about just on the read path?